### PR TITLE
kconf 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by particledecay**
Adding the formula for the [kconf](https://github.com/particledecay/kconf) tool, for managing multiple Kubernetes files. This is my first Homebrew formula and [someone requested this be in Homebrew](https://github.com/particledecay/kconf/issues/50), so I'm happy to make any changes if I got this wrong.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/k/kconf.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Kconf < Formula
+  desc "Manage multiple kubeconfigs easily"
```
Comment (p-linnane) [inline Formula/k/kconf.rb:2]:
```suggestion
  desc "CLI for managing multiple kubeconfigs"
```

File: Formula/k/kconf.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://github.com/particledecay/kconf"
+  url "https://github.com/particledecay/kconf/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "e7fe6750c719caf227890a31e2fd5446972ec3e1e5492180d0a387fe1c3394c0"
+  license "MIT"
+
```
Comment (p-linnane) [inline Formula/k/kconf.rb:7]:
```suggestion
head "https://github.com/particledecay/kconf.git", branch: "main"

```

File: Formula/k/kconf.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    system "go", "mod", "download"
+    system "go", "build", "-ldflags", "-s -w -X github.com/particledecay/kconf/build.Version=#{version} -X github.com/particledecay/kconf/build.Commit=b1261a8-homebrew -X github.com/particledecay/kconf/build.Date=#{Time.now.utc.iso8601}", "-o", "#{bin}/kconf", "."
```
Comment (p-linnane) [inline Formula/k/kconf.rb:13]:
```suggestion
    ldflags = %W[
      -s -w
      -X github.com/particledecay/kconf/build.Version=#{version}
      -X github.com/particledecay/kconf/build.Commit=Homebrew
      -X github.com/particledecay/kconf/build.Date=#{time.iso8601}"
    ]

    system "go", "build", *std_go_args(ldflags:)

    generate_completions_from_executable(bin/"kconf", "completion")
```

File: Formula/k/kconf.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  test do
+    assert_match "v#{version}", shell_output("#{bin}/kconf version")
+    assert_match(/SHA:.*homebrew$/, shell_output("#{bin}/kconf -v version"))
```
Comment (p-linnane) [inline Formula/k/kconf.rb:18]:
```suggestion
    assert_match version.to_s, shell_output(bin/"kconf version")

    output = shell_output(bin/"kconf namespace homebrew 2>&1", 1)
    expected = "you must first set a current context before setting a preferred namespace"
    assert_match expected, output
```

**Review by p-linnane — CHANGES_REQUESTED** [review]
Thanks for the PR @particledecay. We normally do not allow self-submissions from software authors, but I'm willing to make an exception here since I can see the value of distributing `kconf`.

I've provided a few suggestions to align this with our usual syntax and standards. Please let me know if you have any concerns.

**Review by p-linnane — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@particledecay, thanks for your first contribution to homebrew-core!!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**